packages/integration-tests/src/test-project-setup/passwordless_auth.ts

```diff
@@ -0,0 +1,92 @@
+import fs from 'fs/promises';
+import { CloudFormationClient } from '@aws-sdk/client-cloudformation';
+import { TestProjectBase } from './test_project_base.js';
+import { TestProjectCreator } from './test_project_creator.js';
+import { CognitoIdentityProviderClient } from '@aws-sdk/client-cognito-identity-provider';
+import { createEmptyAmplifyProject } from './create_empty_amplify_project.js';
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import { DeployedResourcesFinder } from '../find_deployed_resource.js';
+import assert from 'node:assert';
+
+/**
+ * Creates test projects with passwordless auth.
+ */
+export class PasswordlessAuthTestProjectCreator implements TestProjectCreator {
+  readonly name = 'passwordless-auth';
+
+  /**
+   * Creates project creator.
+   */
+  constructor(
+    private readonly cfnClient: CloudFormationClient,
+    private readonly cognitoIdentityProviderClient: CognitoIdentityProviderClient,
+    private readonly resourceFinder: DeployedResourcesFinder
+  ) {}
+
+  createProject = async (e2eProjectDir: string): Promise<TestProjectBase> => {
+    const { projectName, projectRoot, projectAmplifyDir } =
+      await createEmptyAmplifyProject(this.name, e2eProjectDir);
+
+    const project = new PasswordlessAuthTestProject(
+      projectName,
+      projectRoot,
+      projectAmplifyDir,
+      this.cfnClient,
+      this.cognitoIdentityProviderClient,
+      this.resourceFinder
+    );
+    await fs.cp(
+      project.sourceProjectAmplifyDirPath,
+      project.projectAmplifyDirPath,
+      {
+        recursive: true,
+      }
+    );
+    return project;
```

Comment:
Is there an opportunity to pull any of this into a super method impl that is shared across project creators?